var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Survey(CMakePackage):
+    """Survey is a high-level performance tool product from Trenza, Inc.
+       The survey collector/analytics framework is a new generation,
+       high-level, light-weight multi-platform Linux tool set that
+       targets metric collection for high-level performance analysis
+       of applications running on both single node and on large-scale
+       platforms, including the Cray platforms.
+
+       The collector is designed to work on sequential, MPI, OpenMP,
+       and hybrid codes and directly leverages several interfaces
+       available for tools inside current MPI implementations including:
+       MPICH, MVAPICH, MPT, and OpenMPI. It also supports multiple
+       architectures and has been tested on machines based on Intel,
+       AMD, ARM, and IBM P8/9 processors and integrated GPUs.
+    """
+
+    # To access the survey source and build with
+    # spack please contact: Trenza Inc. via:
+    # dmont@trenzasynergy.com
+    homepage = "http://www.trenzasynergy.com"
+    git      = "git@gitlab.com:trenza/survey.git"
+
+    maintainers = ['jgalarowicz']
+
+    version('develop', branch='master')
+    version('1.0.0', branch='1.0.0')
+
+    variant('mpi', default=False,
+            description="Enable mpi, build MPI data collector")
+
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo'))
+
+    # must have cmake at 3.12 or greater to find python3
+    depends_on('cmake@3.12:')
+
+    # for collectors
+    depends_on("libmonitor@2021.04.27+commrank", type=('build', 'link', 'run'))
```

Comment:
Does `libmonitor` provide executables that are used at runtime? I would've expected a library dependency to be a `'link'` or `('build', 'link')` type dependency, not a run dep.

Reply:
Yes, there are a couple of executables that we can use:
```
galarowi:onyx02.erdc.hpc.mil% module show libmonitor-2021.04.27-gcc-10.2.0-xku5vka
-------------------------------------------------------------------
/p/home/galarowi/spack/share/spack/modules/cray-sles15-broadwell/libmonitor-2021.04.27-gcc-10.2.0-xku5vka:

module-whatis	 Libmonitor is a library providing callback functions for the begin and end of processes and threads. It provides a layer on which to build process monitoring tools and profilers. 
prepend-path	 LD_LIBRARY_PATH /p/home/galarowi/spack/opt/spack/cray-sles15-broadwell/gcc-10.2.0/libmonitor-2021.04.27-xku5vkajk7eogf5thlcyhqeiw3urir4k/lib 
prepend-path	 PATH /p/home/galarowi/spack/opt/spack/cray-sles15-broadwell/gcc-10.2.0/libmonitor-2021.04.27-xku5vkajk7eogf5thlcyhqeiw3urir4k/bin 
prepend-path	 CMAKE_PREFIX_PATH /p/home/galarowi/spack/opt/spack/cray-sles15-broadwell/gcc-10.2.0/libmonitor-2021.04.27-xku5vkajk7eogf5thlcyhqeiw3urir4k/ 
-------------------------------------------------------------------

galarowi:onyx02.erdc.hpc.mil% lsr /p/home/galarowi/spack/opt/spack/cray-sles15-broadwell/gcc-10.2.0/libmonitor-2021.04.27-xku5vkajk7eogf5thlcyhqeiw3urir4k/bin
total 16
4 -rwxr-xr-x 1 galarowi 3414L200 2028 Aug 31 15:04 monitor-run
4 -rwxr-xr-x 1 galarowi 3414L200 3511 Aug 31 15:04 monitor-link
4 drwxr-sr-x 6 galarowi 3414L200 4096 Aug 31 15:04 ..
4 drwxr-sr-x 2 galarowi 3414L200 4096 Aug 31 15:04 .
```